core/src/main/java/org/apache/iceberg/RemoveSnapshots.java

```diff
@@ -138,10 +138,23 @@ public ExpireSnapshots executeDeleteWith(ExecutorService executorService) {
   private TableMetadata internalApply() {
     this.base = ops.refresh();
 
-    return base.removeSnapshotsIf(snapshot ->
+    TableMetadata updated = base.removeSnapshotsIf(snapshot ->
         idsToRemove.contains(snapshot.snapshotId()) ||
         (expireOlderThan != null && snapshot.timestampMillis() < expireOlderThan &&
             !idsToRetain.contains(snapshot.snapshotId())));
+
+    List<Snapshot> updateSnapshots = updated.snapshots();
+    List<Snapshot> baseSnapshots = base.snapshots();
+    if (updateSnapshots.size() != baseSnapshots.size()) {
+      return updated;
+    } else {
+      for (int i = 0; i < baseSnapshots.size(); i += 1) {
+        if (!updateSnapshots.get(i).equals(baseSnapshots.get(i))) {
+          return updated;
+        }
```

Comment:
I don't think this block is needed. The contract of `removeSnapshotsIf` is to only remove snapshots or return the list unmodified. We can rely on that and just check that the size hasn't changed.